### ISSUE TITLE
Port multi-gpu support

### ISSUE THREAD

**Issue by glinscott** [glinscott/leela-chess#38]
https://github.com/gcp/leela-zero/commit/5855e0fca6713d3f75ec6d13088cc0c3826908a7

**Comment by vondele** [comment]
@glinscott @gcp in the longer run it would be ideal if both projects could somehow share a common infrastructure (essentially a library or so), well separated from the game specific parts. Not sure how feasible this is, but it would reduce the number of things that need to be ported.

**Comment by glinscott** [comment]
Yes, agreed, they are moving pretty fast over there, and it's tough to keep up :).

**Comment by gcp** [comment]
I think the most interesting parts for other projects are
UCTSearch
UCTNode
OpenCL
OpenCLScheduler
Network
Tuning

and utilities like Utils/ThreadPool/NNCache/TTable

Maybe one possible start would be to subclass UCTSearch/UCTNode and put game specific logic in the subclasses. And maybe not hardcode the board size, which also annoys the people doing 9x9 go.
